src/Splitio/Services/Client/Classes/SplitClient.cs

```diff
@@ -434,26 +435,46 @@ protected void BuildFlagSetsFilter(HashSet<string> sets)
         #region Private Async Methods
         private async Task<List<TreatmentResult>> GetTreatmentsAsync(Enums.API method, Key key, List<string> features, Dictionary<string, object> attributes)
         {
-            features = _clientExtensionService.TreatmentsValidations(method, key, features, _log, out List<TreatmentResult> controlTreatments);
+            try
+            {
+                features = _clientExtensionService.TreatmentsValidations(method, key, features, _log, out List<TreatmentResult> controlTreatments);
 
-            if (controlTreatments != null) return controlTreatments;
+                if (controlTreatments != null) return controlTreatments;
 
-            var treatments = await _evaluator.EvaluateFeaturesAsync(method, key, features, attributes);
+                var treatments = await _evaluator.EvaluateFeaturesAsync(method, key, features, attributes);
 
-            await TrackImpressionsAsync(treatments, key);
+                await TrackImpressionsAsync(treatments, key);
+
+                return treatments;
+            }
+            catch (Exception ex)
+            {
+                _telemetryEvaluationProducer.RecordException(method.ConvertToMethodEnum());
 
-            return treatments;
+                _log.Warn("Something went wrong evaluating features.", ex);
```

Comment:
Just wondering if it should be an error log, like when calling the track method:
```
_log.Error("Exception caught trying to track an event", e);
```